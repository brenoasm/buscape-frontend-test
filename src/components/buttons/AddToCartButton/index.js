import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import { colors } from '../../../theme';

const propTypes = {
  handleProductAddToCartClick: PropTypes.func,
  product: PropTypes.object
};

const defaultProps = {
  handleProductAddToCartClick: () => {},
  product: null
};

const StyledAddToCartButton = styled.div`
  color: ${colors.white};
  background-color: ${colors.green};
  border: none;
  border-radius: 2px;
  padding: 0 10px;
  cursor: pointer;

  > span {
    text-align: center;
    font-size: 14px;
  }
`;

const AddToCartButton = ({ handleProductAddToCartClick, product }) => (
  <StyledAddToCartButton onClick={() => handleProductAddToCartClick(product)}>
    <span>
      Adicionar ao carrinho <i className="fas fa-chevron-right"></i>
    </span>
  </StyledAddToCartButton>
);

AddToCartButton.propTypes = propTypes;
AddToCartButton.defaultProps = defaultProps;

export default AddToCartButton;
