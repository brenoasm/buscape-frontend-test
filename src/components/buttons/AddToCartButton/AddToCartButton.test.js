import React from 'react';
import { shallow } from 'enzyme';

import AddToCartButton from './';

describe('AddToCartButton', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<AddToCartButton />);

    expect(wrapper).toMatchSnapshot();
  });

  it('calls the fn when clicked', () => {
    const fn = jest.fn();
    const wrapper = shallow(<AddToCartButton handleProductAddToCartClick={fn} />);
    wrapper.simulate('click');

    expect(fn).toBeCalled();
  });
});
