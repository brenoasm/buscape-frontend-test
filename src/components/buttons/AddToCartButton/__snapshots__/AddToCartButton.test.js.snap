// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToCartButton matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddToCartButton
    handleProductAddToCartClick={[Function]}
    product={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <span>
        Adicionar ao carrinho 
        <i
          className="fas fa-chevron-right"
        />
      </span>,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Adicionar ao carrinho ",
          <i
            className="fas fa-chevron-right"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Adicionar ao carrinho ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fas fa-chevron-right",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
      ],
      "type": "span",
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  color: ",
          "#FFF",
          ";
  background-color: ",
          "#22B14C",
          ";
  border: none;
  border-radius: 2px;
  padding: 0 10px;
  cursor: pointer;

  > span {
    text-align: center;
    font-size: 14px;
  }
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <span>
          Adicionar ao carrinho 
          <i
            className="fas fa-chevron-right"
          />
        </span>,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Adicionar ao carrinho ",
            <i
              className="fas fa-chevron-right"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Adicionar ao carrinho ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-chevron-right",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": "span",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  color: ",
            "#FFF",
            ";
  background-color: ",
            "#22B14C",
            ";
  border: none;
  border-radius: 2px;
  padding: 0 10px;
  cursor: pointer;

  > span {
    text-align: center;
    font-size: 14px;
  }
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
